api/src/main/java/org/openmrs/api/db/ConceptDAO.java

```diff
@@ -84,42 +83,74 @@
 	public Concept getConcept(Integer conceptId) throws DAOException;
 	
 	/**
+	 * Get the concept name
+	 * 
 	 * @see org.openmrs.api.ConceptService#getConceptName(java.lang.Integer)
-	 * @param conceptNameId
-	 * @return The ConceptName matching the specified conceptNameId
+	 * @param conceptNameId The concept name id
+	 * @return The concept name matching the specified conceptNameId
+	 * @throws DAOException
 	 */
 	public ConceptName getConceptName(Integer conceptNameId) throws DAOException;
 	
 	/**
+	 * Returns all concepts
+	 * 
 	 * @see org.openmrs.api.ConceptService#getAllConcepts(java.lang.String, boolean, boolean)
+	 * @throws DAOException
 	 */
 	public List<Concept> getAllConcepts(String sortBy, boolean asc, boolean includeRetired) throws DAOException;
 	
 	/**
 	 * Returns a list of concepts based on the search criteria
 	 * 
-	 * @param name
-	 * @param loc
+	 * @param name The name of the concept to search
+	 * @param loc The locale to look in
 	 * @param searchOnPhrase This puts wildcard characters around the concept name search criteria
-	 * @return List&lt;Concept&gt;
+	 * @return a list of concepts based on search criteria
 	 * @throws DAOException
 	 * <strong>Should</strong> not return concepts with matching names that are voided
 	 */
 	public List<Concept> getConcepts(String name, Locale loc, boolean searchOnPhrase, List<ConceptClass> classes,
 	        List<ConceptDatatype> datatypes) throws DAOException;
 	
 	/**
+	 * Return a list of concept results based on search criteria
+	 * 
 	 * @see ConceptService#getConcepts(String, List, boolean, List, List, List, List, Concept,
 	 *      Integer, Integer)
+	 * @param phrase 
```

Comment:
Shouldn't we have some descriptions for these parameters?

Reply:
I  actually needed some one well informed about this method , to give me the details of what the params do . Any info on this @ibacher ?